### PR TITLE
More rules for DE sites

### PR DESCRIPTION
**Pull request by jdorweiler**
Adding more coverage for DE sites. 
- Most test cases are the 'name' field in the file
- Exceptions are for larger CMPs
    - Borlabs: https://reitschuster.de/
    - Usercentrics: https://dm.de https://hornbach.de

### CODE REVIEW

File: rules/autoconsent/hornbach-de.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "hornbach-de",
+  "detectCmp": [{ "eval": "!!document.querySelector('#usercentrics-root').shadowRoot.querySelector('#uc-center-container')"}],
```
Comment (muodov) [inline rules/autoconsent/hornbach-de.json:3]: Judging by the class names, it's [usercentrics](https://usercentrics.com/), should we have a more generic solution?

Comment (jdorweiler) [reply rules/autoconsent/hornbach-de.json:3]: 👍 nice find,  I should for sure rename it. I'll have to see if we run into more example sites to make a general case.

**Review by muodov — COMMENTED** [review]
So I found that many of these rules (I've marked only some of them) will stop working after the fix in https://github.com/duckduckgo/autoconsent/pull/4 - basically, [this bug](https://app.asana.com/0/1201844467387842/1202153914495783)
I don't think it should block this PR though, I'd rather fix that as part of https://github.com/duckduckgo/autoconsent/pull/4, after you merge this.

**Comment by muodov** [comment]
Also, do you want to add the "test" entries to help the automated crawler?

File: rules/autoconsent/hornbach-de.json (hunk 1 above)
Comment (adewes) [reply rules/autoconsent/hornbach-de.json]: Would it make sense to restrict these rules to specific websites? In my understanding some of them are one-off fixes for these websites, so we shouldn't execute them on other pages at all. I see that there's a `UrlRule` type but I don't think it's used in the CMP logic as of now. Maybe we could add logic to ensure site-specific rules are only executed on their respective domains? That could make rule execution faster for other sites as well.

Comment (jdorweiler) [reply rules/autoconsent/hornbach-de.json]: This ended up being a bigger CMP. I moved it over to https://github.com/duckduckgo/autoconsent/pull/3/files#diff-b383aa32dd7c9e721949d46b3424e4076abd6e2eb0b74efae275c6f35630d102. 

I'll make a followup for the UrlRule.

File: rules/autoconsent/springer.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "springer.com",
+  "detectCmp": [{ "exists": ".cc-banner"}],
```
[the remark this answers is not in the document]
Comment (jdorweiler) [reply rules/autoconsent/springer.json:3]: I'll just drop this one. I don't even see a cookie popup on the site anymore.

File: rules/autoconsent/borlabs.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
```
[the remark this answers is not in the document]
Comment (jdorweiler) [reply rules/autoconsent/borlabs.json:1]: I think I'd rather leave this one until we get more examples of their popup.

**Pull request merged**